lib/jekyll/tags/include.rb

```diff
@@ -253,20 +253,20 @@ def tag_includes_dirs(context)
       end
 
       def page_path(context)
-        if context.registers[:page].nil?
-          context.registers[:site].source
-        else
-          site = context.registers[:site]
-          page_payload  = context.registers[:page]
-          resource_path = \
-            if page_payload["collection"].nil?
-              page_payload["path"]
-            else
-              File.join(site.config["collections_dir"], page_payload["path"])
-            end
-          resource_path.sub!(%r!/#excerpt\z!, "")
-          site.in_source_dir File.dirname(resource_path)
-        end
+        page = context.registers[:page]
+        site = context.registers[:site]
+        return site.source unless page
+
+        path = resource_path(page, site)
+        site.in_source_dir File.dirname(path)
```

Comment:
The local variable assignment can be done away with..
```suggestion
        site.in_source_dir File.dirname(resource_path(page, site))
```

Reply:
Sometimes it improves readability. Although I could do it like this:
```
directory_path = File.dirname resource_path(page, site)
site.in_source_dir directory_path
```